fvtr/systemtap/systemtap.exp

```diff
@@ -49,17 +49,21 @@ if { $env(AT_CROSS_BUILD) == "yes" } {
 	set install_prefix $env(AT_DEST)
 }
 
+# Starting with glibc 2.34 (AT 15.0) libpthread was moved into libc.so
 if { $TARGET32 } {
-	set lib [glob $install_prefix/lib/libpthread-*.so]
-	if {[check_systemtap $readelf_exe $lib]} {
-		set ret 1
-	}
+	set libpath "$install_prefix/lib"
+} else {
+	set libpath "$install_prefix/lib64"
 }
-if { $TARGET64 } {
-	set lib [glob $install_prefix/lib64/libpthread-*.so]
-	if {[check_systemtap $readelf_exe $lib]} {
-		set ret 1
-	}
+
+if { $::env(AT_MAJOR_VERSION) < 15.0 } {
+	set lib [glob $libpath/libpthread-*.so]
+} else {
+	set lib [glob $libpath/libc-*.so]
+}
+
+if {[check_systemtap $readelf_exe $lib]} {
```

Comment:
I don't think these changes are correct because they assume a single libc or libpthread is available.
In a big endian system, AT might provide both 32 and 64 bits libraries.  That means the test has to be executed twice.

Reply:
True. I changed the PR to maintain the same test structure as before (guaranteeing the test will run twice, for 32-bit and 64-bit).